File: problems/89-gray-code/gray-code.java
class Solution {
    public List<Integer> grayCode(int n) {
        int len = (int) Math.pow(2, n);
        List<Integer> arr = new ArrayList<>();
        for (int i = 0; i < len; i++) {
            arr.add(i ^ (i >> 1));
        }
        
        return arr;
    }
}

File: problems/89-gray-code/gray-code.cpp
class Solution {
public:
    vector<int> grayCode(int n) {
        long long len = pow(2, n);
        vector<int> arr;
        for (int i = 0; i < len; i++) {
            arr.push_back(i ^ (i >> 1));
        }
        return arr;
    }
};